The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3154163025/jobs/5131451032) [(plain)](https://github.com/rust-lang/rust/commit/76105997738d262c098ebba221c124cdd73dddf1/checks/8626489850/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking rustc_ast_passes v0.0.0 (/checkout/compiler/rustc_ast_passes)
error[E0432]: unresolved import `rustc_macros::SessionDiagnostic`
 --> compiler/rustc_middle/src/error.rs:1:32
  |
1 | use rustc_macros::{Diagnostic, SessionDiagnostic};
  |                                ^^^^^^^^^^^^^^^^^ no `SessionDiagnostic` in the root

error: cannot determine resolution for the derive macro `SessionDiagnostic`
   |
52 | #[derive(SessionDiagnostic)]
   |          ^^^^^^^^^^^^^^^^^
   |
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `diag` in this scope
   |
   |
53 | #[diag(middle::const_eval_non_int)]

error: cannot find attribute `primary_span` in this scope
  --> compiler/rustc_middle/src/error.rs:55:7
   |
   |
55 |     #[primary_span]

    Checking rustc_expand v0.0.0 (/checkout/compiler/rustc_expand)
    Checking rustc_builtin_macros v0.0.0 (/checkout/compiler/rustc_builtin_macros)
error[E0308]: mismatched types
error[E0308]: mismatched types
   --> compiler/rustc_middle/src/ty/adt.rs:461:76
    |
461 |                     tcx.sess.emit_err(crate::error::ConstEvalNonIntError { span });
    |                                                                            ^^^^ expected struct `rustc_span::Span`, found struct `TyCtxtAt`

error[E0277]: the trait bound `ConstEvalNonIntError: IntoDiagnostic<'_>` is not satisfied
   --> compiler/rustc_middle/src/ty/adt.rs:461:39
    |
461 |                     tcx.sess.emit_err(crate::error::ConstEvalNonIntError { span });
    |                              -------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `IntoDiagnostic<'_>` is not implemented for `ConstEvalNonIntError`
    |                              required by a bound introduced by this call
    |
    |
note: required by a bound in `Session::emit_err`
    |
    |
480 |     pub fn emit_err<'a>(&'a self, err: impl IntoDiagnostic<'a>) -> ErrorGuaranteed {
    |                                             ^^^^^^^^^^^^^^^^^^ required by this bound in `Session::emit_err`
Some errors have detailed explanations: E0277, E0308, E0432.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `rustc_middle` due to 6 previous errors
warning: build failed, waiting for other jobs to finish...
```

</details>

--- a/compiler/rustc_error_messages/locales/en-US/middle.ftl
+++ b/compiler/rustc_error_messages/locales/en-US/middle.ftl
@@ -15,3 +15,6 @@ middle_previous_use_here =
 middle_limit_invalid =
     `limit` must be a non-negative integer
     .label = {$error_str}
+
+middle_const_eval_non_int =
+    constant evaluation of enum discriminant resulted in non-integer

--- a/compiler/rustc_middle/src/error.rs
+++ b/compiler/rustc_middle/src/error.rs
@@ -48,3 +48,10 @@ pub struct LimitInvalid<'a> {
     pub value_span: Span,
     pub error_str: &'a str,
 }
+
+#[derive(Diagnostic)]
+#[diag(middle::const_eval_non_int)]
+pub struct ConstEvalNonIntError {
+    #[primary_span]
+    pub span: Span,
+}

--- a/compiler/rustc_middle/src/ty/adt.rs
+++ b/compiler/rustc_middle/src/ty/adt.rs
@@ -458,11 +458,9 @@ impl<'tcx> AdtDef<'tcx> {
                     Some(Discr { val: b, ty })
                 } else {
                     info!("invalid enum discriminant: {:#?}", val);
-                    crate::mir::interpret::struct_error(
-                        tcx.at(tcx.def_span(expr_did)),
-                        "constant evaluation of enum discriminant resulted in non-integer",
-                    )
-                    .emit();
+                    tcx.sess.emit_err(crate::error::ConstEvalNonIntError {
+                        span: tcx.def_span(expr_did),
+                    });
                     None
                 }
             }

